integration_test/ops_agent_test.go

```diff
@@ -1350,22 +1264,40 @@ func testDefaultMetrics(ctx context.Context, t *testing.T, logger *logging.Direc
 		}
 	}
 
-	// First make sure that the uptime metrics are being uploaded.
-	var uptimeWaitGroup sync.WaitGroup
-	regexes := agentVersionRegexesForPlatform(vm.Platform)
-	for _, versionRegex := range regexes {
-		versionRegex := versionRegex
-		uptimeWaitGroup.Add(1)
-		go func() {
-			defer uptimeWaitGroup.Done()
-			if _, err := gce.WaitForMetric(ctx, logger.ToMainLog(), vm, "agent.googleapis.com/agent/uptime", window,
-				[]string{fmt.Sprintf("metric.labels.version = monitoring.regex.full_match(%q)", versionRegex)},
-			); err != nil {
-				t.Error(err)
-			}
-		}()
+	bytes, err := os.ReadFile(path.Join("agent_metrics", "metadata.yaml"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var agentMetrics struct {
+		ExpectedMetrics []*common.ExpectedMetric `yaml:"expected_metrics" validate:"onetrue=Representative,unique=Type,dive"`
+	}
```

Comment:
[Optional] To avoid repetition, I wonder if it makes sense to define a `common.ExpectedMetrics` struct for this and embed it in `IntegrationMetadata`? For example:
```go
type ExpectedMetrics struct {
	ExpectedMetrics              []*ExpectedMetric            `yaml:"expected_metrics" validate:"onetrue=Representative,unique=Type,dive"`
}

type IntegrationMetadata struct {
	PublicUrl                    string                       `yaml:"public_url"`
	AppUrl                       string                       `yaml:"app_url" validate:"required,url"`
	ShortName                    string                       `yaml:"short_name" validate:"required,excludesall=‘’“”"`
	LongName                     string                       `yaml:"long_name" validate:"required,excludesall=‘’“”"`
	LogoPath                     string                       `yaml:"logo_path"`
	Description                  string                       `yaml:"description" validate:"required,excludesall=‘’“”"`
	ConfigurationOptions         *ConfigurationOptions        `yaml:"configuration_options" validate:"required"`
	ConfigureIntegration         string                       `yaml:"configure_integration"`
	ExpectedLogs                 []*ExpectedLog               `yaml:"expected_logs" validate:"dive"`
	ExpectedMetrics              ExpectedMetrics              `yaml:",inline" validate:"dive"`
	MinimumSupportedAgentVersion MinimumSupportedAgentVersion `yaml:"minimum_supported_agent_version"`
	SupportedAppVersion          []string                     `yaml:"supported_app_version" validate:"required,unique,min=1"`
	SupportedOperatingSystems    string                       `yaml:"supported_operating_systems" validate:"required,oneof=linux windows linux_and_windows"`
	RestartAfterInstall          bool                         `yaml:"restart_after_install"`
	Troubleshoot                 string                       `yaml:"troubleshoot" validate:"excludesall=‘’“”"`
}
```
For symmetry, you may choose to do the same for `ExpectedLogs` as well…

Reply:
deferred to #732